tapdb/universe.go

```diff
@@ -64,6 +64,15 @@ type BaseUniverseStore interface {
 	QueryUniverseLeaves(ctx context.Context,
 		arg UniverseLeafQuery) ([]UniverseLeaf, error)
 
+	// DeleteUniverseLeaves is used to delete leaves that reside in a
+	// universe tree.
+	DeleteUniverseLeaves(ctx context.Context, namespace string) error
```

Comment:
See prior comment, I think we can ideally lean on the `ON DELETE CASCADE` behavior.